shared/actions/chat/inbox.js

```diff
@@ -2,141 +2,279 @@
+    if (selectedConversation === inboxState.get('conversationIDKey')) {
+      // load validated selected
+      yield put(Creators.loadMoreMessages(selectedConversation, false))
+    }
+  }
+}
+
+// Gui is showing boxed content, find some rows to unbox
+function * untrustedInboxVisible (action: Constants.UntrustedInboxVisible): SagaGenerator<any, any> {
+  const {conversationIDKey, rowsVisible} = action.payload
+  const inboxes = yield select(state => state.chat.get('inbox'))
+
+  const idx = inboxes.findIndex(inbox => inbox.conversationIDKey === conversationIDKey)
+  if (idx === -1) {
+    return
+  }
+
+  // Collect items to unbox
+  const total = rowsVisible * 2
+  const conversationIDKeys = inboxes.slice(idx, idx + total).map(i => i.state === 'untrusted' ? i.conversationIDKey : null).filter(Boolean).toArray()
+
+  if (conversationIDKeys.length) {
+    yield call(_unboxConversations, conversationIDKeys)
+  }
+}
+
+// Loads the trusted inbox segments
+function * _unboxConversations (conversationIDKeys: Array<Constants.ConversationIDKey>): Generator<any, any, any> {
+  yield put(Creators.setUnboxing(conversationIDKeys))
+
+  const channelConfig = singleFixedChannelConfig([
+    'chat.1.chatUi.chatInboxUnverified',
+    'chat.1.chatUi.chatInboxConversation',
+    'chat.1.chatUi.chatInboxFailed',
+    'finished',
+  ])
+
+  const loadInboxChanMap: ChannelMap<any> = ChatTypes.localGetInboxNonblockLocalRpcChannelMap(channelConfig, {
+    param: {
+      identifyBehavior: TlfKeysTLFIdentifyBehavior.chatGui,
+      query: {
+        ..._getInboxQuery,
+        convIDs: conversationIDKeys.map(Constants.keyToConversationID),
+      },
+    },
+  })
+
+  while (true) {
+    const incoming: {[key: string]: any} = yield race({
+      chatInboxConversation: takeFromChannelMap(loadInboxChanMap, 'chat.1.chatUi.chatInboxConversation'),
+      chatInboxFailed: takeFromChannelMap(loadInboxChanMap, 'chat.1.chatUi.chatInboxFailed'),
+      chatInboxUnverified: takeFromChannelMap(loadInboxChanMap, 'chat.1.chatUi.chatInboxUnverified'),
+      finished: takeFromChannelMap(loadInboxChanMap, 'finished'),
+      timeout: call(delay, 30000),
+    })
+
+    // Ignore untrusted version
+    if (incoming.chatInboxUnverified) {
+      incoming.chatInboxUnverified.response.result()
+    } else if (incoming.chatInboxConversation) {
+      requestIdleCallback(() => { incoming.chatInboxConversation.response.result() }, {timeout: 100})
+      yield call(delay, 1)
+      yield call(updateInbox, incoming.chatInboxConversation.params.conv)
       // find it
     } else if (incoming.chatInboxFailed) {
-      console.log('ignoring chatInboxFailed', incoming.chatInboxFailed)
-      requestIdleCallback(() => {
-        incoming.chatInboxFailed.response.result()
-      }, {timeout: 100})
-
+      console.log('chatInboxFailed', incoming.chatInboxFailed)
+      requestIdleCallback(() => { incoming.chatInboxFailed.response.result() }, {timeout: 100})
       yield call(delay, 1)
```

Comment:
This is existing behavior. It gives us some breathing room from a cpu perspective as we don't control the rate of these things coming in